### PR TITLE
feat: added settings command

### PR DESCRIPTION
**Pull request by KommentatorForAll**

### CODE REVIEW

File: src/modules/interactions/buttons/settings.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+ *
+ * @param {import('discord.js').Client} client
+ * @param {import('discord.js').ButtonInteraction} interaction
+ * @param {string[]} args
+ */
+async function fun(client, interaction, args) {
+    const page = EmbedPageMessage.getInstance(interaction.message.id);
+    if (!page) {
+        await interaction.update({
+            content: 'This settings has expired. Please run `settings` again!',
```
Comment (Chicken) [inline src/modules/interactions/buttons/settings.js:70]:
```suggestion
            content: 'These settings have expired. Please run `settings` again!',
```

File: src/modules/interactions/buttons/settings.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        page.index = (page.index + 1) % page.pages.length;
+    } else {
+        switch (args[0]) {
+            case 'os':
+                await handleOS(await getMember(interaction.user.id), currentPage, args[1]);
+                break;
+            default:
+                await handleToggle(await getMember(interaction.user.id), currentPage, args[0]);
+        }
+    }
```
Comment (Chicken) [inline src/modules/interactions/buttons/settings.js:92]:
```suggestion
    switch (args[0]) {
        case 'left':
           page.index = (page.index + pages.pages.length - 1) % pages.pages.length;
           break;
        case 'right':
           page.index = (page.index + 1) % page.pages.length;
           break;
        case 'os':
           await handleOS(await getMember(interaction.user.id), currentPage, args[1]);
           break;
        default:
           await handleToggle(await getMember(interaction.user.id), currentPage, args[0]);
    }
```

File: src/modules/interactions/buttons/settings.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    await page.update();
+    await interaction.deferUpdate();
```
Comment (Chicken) [inline src/modules/interactions/buttons/settings.js:94]: Um no. That will just fail after 15 minutes I believe. Instead the update method of the pagination should take in the interaction as argument and call update on the interaction with the edited message

File: src/modules/dmCommands/commands/settings.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+/**
+ *
+ * @param {import('discord.js').Client} client
+ * @param {import('discord.js').Message} message
+ * @return {Promise<void>}
+ */
+export async function fun(client, message) {
+    const embedPages = await getPages(message.author);
+    const embedPageMessage = new EmbedPageMessage(embedPages);
+    await embedPageMessage.update(message);
```
Comment (Chicken) [inline src/modules/dmCommands/commands/settings.js:102]: Maybe just take in the message at constructor so no need to call update.

File: src/dto/EmbedPageMessage.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+     */
+    async send(message = null) {
+        this.message = await message.reply('settings');
+        cache[this.message.id] = this;
+    }
+
+    async update(message) {
+        if (!this.message) {
+            await this.send(message);
+        }
```
Comment (Chicken) [inline src/dto/EmbedPageMessage.js:35]: Continuation of some other reviews, send the message in constructor and then make update take in interaction.

**Review by Chicken — CHANGES_REQUESTED** [review]

File: src/modules/dmCommands/commands/settings.js [hunk 6]
```diff
@@ -0,0 +1,103 @@
+import { MessageButton } from 'discord.js';
+import EmbedPage from '../../../dto/EmbedPage';
+import { getMember, getRole, userDb } from '../../../util';
+import EmbedPageMessage from '../../../dto/EmbedPageMessage';
+
+export const command = 'settings';
+export const desc = 'Configure your profile';
```
Comment (WilsontheWolf) [inline src/modules/dmCommands/commands/settings.js:7]: Needs punctuation to be constant, Also your profile, is kinda vague, Maybe more like 
```suggestion
export const desc = 'Configure your user settings.';
```

File: src/dto/EmbedPageMessage.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        const page = this.pages[this.index];
+        this.message.edit({
+            embeds: [page.toEmbed()],
+            components: [
+                new MessageActionRow().addComponents(page.buttons),
+                new MessageActionRow().addComponents([
+                    new MessageButton()
+                        .setLabel('«')
+                        .setStyle('PRIMARY')
+                        .setCustomId(`settings.left`),
```
Comment (WilsontheWolf) [inline src/dto/EmbedPageMessage.js:45]: Maybe calculate the page id here and make it simpler on yourself later

File: src/modules/interactions/buttons/settings.js (hunk 1 above)
Comment (WilsontheWolf) [inline src/modules/interactions/buttons/settings.js:70]: Is this necessary? Could we not maybe make a new page to continue on from.

File: src/modules/interactions/buttons/settings.js (hunk 2 above)
Comment (WilsontheWolf) [inline src/modules/interactions/buttons/settings.js:92]: Mixing if and switch
```suggestion
switch (args[0]) {
    case 'left':
        page.index -= 1;
        if (page.index < 0) {
            page.index = page.pages.length - 1;
        }
        break;
    case 'right':
        page.index = (page.index + 1) % page.pages.length;
        break;
    case 'os':
        await handleOS(await getMember(interaction.user.id), currentPage, args[1]);
        break;
     default:
        await handleToggle(await getMember(interaction.user.id), currentPage, args[0]);
    }
```

**Review by WilsontheWolf — CHANGES_REQUESTED** [review]
Just a first review. Will check more later

File: src/modules/interactions/buttons/settings.js (hunk 2 above)
Comment (Chicken) [reply src/modules/interactions/buttons/settings.js:92]: _Bruh this is already on my review_

File: src/dto/EmbedPageMessage.js (hunk 7 above)
Comment (Chicken) [reply src/dto/EmbedPageMessage.js:45]: Doesn't really matter where the calculations are done tbh

File: src/modules/interactions/buttons/settings.js (hunk 2 above)
Comment (WilsontheWolf) [reply src/modules/interactions/buttons/settings.js:92]: Well I was making my review before you submitted your review.

Resolving in favour of https://github.com/IIoW/IIslander/pull/185#discussion_r801948134

**Comment by WilsontheWolf** [comment]
https://github.com/IIoW/IIslander/blob/feature/settings/src/constants/Messages.js#L68-L69 needs to be updated

**Comment by WilsontheWolf** [comment]
CODEOWNERS needs to be updated for the new Dto's

File: src/modules/dmCommands/commands/settings.js (hunk 4 above)
Comment (KommentatorForAll) [reply src/modules/dmCommands/commands/settings.js:102]: update is async. afaik a constructor must not be async

Comment (Chicken) [reply src/modules/dmCommands/commands/settings.js:102]: It doesn't need to be async if you don't await the sending

Comment (WilsontheWolf) [reply src/modules/dmCommands/commands/settings.js:102]: Alternitivly pass the messsage to the constructor than make a static prop to create a class where you pass args other than message and then it sends message passes args and message and gives you class

File: src/modules/dmCommands/commands/settings.js (hunk 6 above)
Comment (WilsontheWolf) [reply src/modules/dmCommands/commands/settings.js]: You changed the message here but didn't add puncuation

File: src/dto/EmbedPageMessage.js [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+        });
+    }
+
+    /**
+     *
+     * @param {String} id
+     * @returns {EmbedPageMessage}
+     */
+    static getInstance(id) {
+        return cache[id];
```
Comment (peanutbother) [inline src/dto/EmbedPageMessage.js:61]: Replace with `cache.get(id)`
Maybe add an error check with `cache.has(id)` but depends on where you want to check for existence.

File: src/modules/interactions/buttons/settings.js (hunk 3 above)
Comment (KommentatorForAll) [reply src/modules/interactions/buttons/settings.js:91]: that won't work propperly with the first send of the message

File: src/modules/dmCommands/commands/settings.js (hunk 4 above)
Comment (KommentatorForAll) [reply src/modules/dmCommands/commands/settings.js:102]: by awaiting one ensures that we have a trace of the error if something happens within the await.

File: src/modules/interactions/buttons/settings.js (hunk 1 above)
Comment (KommentatorForAll) [reply src/modules/interactions/buttons/settings.js]: not anymore, as pages have a timeout of 3 minutes now

**Review by Chicken — APPROVED** [review]

**Pull request merged**